libraries/hotstuff/chain_pacemaker.cpp

```diff
@@ -260,23 +260,38 @@ namespace eosio { namespace hotstuff {
       return n;
    }
 
-   std::vector<name> chain_pacemaker::get_finalizers() {
+   std::vector<fc::crypto::blslib::bls_public_key> chain_pacemaker::get_finalizer_keys() {
```

Comment:
Seems rather expensive to copy this on every call. Instead, we should store the active finalizer set in `qc_chain` as that appears to be all that uses it outside of tests. Then we could just have a `get_finalizer_set` that returns `const finalizer_set&`.